tests/distributed_mode/test_single_label_classifier.py

```diff
@@ -0,0 +1,160 @@
+
+    assert hasattr(doc2._, attribute_name)
+    assert doc2.get_attribute(attribute_name) == expected
+
+    # Deploy pipeline 
+    nlp.deploy(worker=bob)
+
+    # Empty local worker's object store to ensure that 
+    # we are using the deployed pipeline rather than local states
+    for obj in list(me._objects):
+        del me._objects[obj]
+
+    # Load deployed pipeline
+    nlp_loaded_from_bob = syfertext.load(pipeline_name)
+
+    # Case 3: infer using local string and deployed pipeline
+    doc3 = nlp_loaded_from_bob(String(text))
+
+    assert hasattr(doc3._, attribute_name)
+    assert doc3.get_attribute(attribute_name) == expected
+
+    # Case 4: infer using string pointer and deployed pipeline
+    string_ptr = String(text).send(bob)
+    doc_ptr = nlp_loaded_from_bob(string_ptr)
+    doc4 = bob._objects[doc_ptr.id_at_location]
+
+    assert hasattr(doc4._, attribute_name)
+    assert doc4.get_attribute(attribute_name) == expected
+
+
+@pytest.mark.parametrize("text,expected", [("The quick brown fox", "B")])
+def test_encrypted_single_label_inference(text, expected):
+    torch.manual_seed(seed)
+    classifier_name = "encrypted_classifier"
+    nlp = get_test_language_model()
+    pipeline_name = nlp.pipeline_name
+
+    # Set up SingleLabelClassifier and add to pipeline
+    net = Net()
+    classifier = SingleLabelClassifier(
+        classifier=net, 
+        doc_encoder=doc_encoder,
+        encryption="mpc",
+        labels=labels
+    )  
+    nlp.add_pipe(classifier, name=classifier_name, access={'*'})
+
+    attribute_name = "{pipeline_name}__{classifier_name}".format(
+        pipeline_name=pipeline_name,
+        classifier_name=classifier_name
+    )
+
+    # Case 1: infer using string and pipeline before deployment
+    doc1 = nlp(String(text))
+
+    assert hasattr(doc1._, attribute_name)
+    assert doc1.get_attribute(attribute_name) == expected
+
+    # Case 2: infer using string pointer and pipeline before deployment
+    string_ptr = String(text).send(bob)
+    doc_ptr = nlp(string_ptr) 
+    doc2 = bob._objects[doc_ptr.id_at_location]
+
+    assert hasattr(doc2._, attribute_name)
+    assert doc2.get_attribute(attribute_name) == expected
+
+    # Deploy pipeline 
+    nlp.deploy(worker=bob)
+
+    # Empty local worker's object store to ensure that 
+    # we are using the deployed pipeline rather than local states
+    for obj in list(me._objects):
+        del me._objects[obj]
+
+    # Load deployed pipeline
+    nlp_loaded_from_bob = syfertext.load(pipeline_name)
+
+    # Case 3: infer using local string and deployed pipeline
+    doc3 = nlp_loaded_from_bob(String(text))
+
+    assert hasattr(doc3._, attribute_name)
+    assert doc3.get_attribute(attribute_name) == expected
+
+    # Case 4: infer using string pointer and deployed pipeline
+    string_ptr = String(text).send(bob)
+    doc_ptr = nlp_loaded_from_bob(string_ptr)
+    doc4 = bob._objects[doc_ptr.id_at_location]
```

Comment:
maybe we can add a case when `String` is at `alice` (say), and the pipeline is deployed at `bob` and `SingleLabelClassifier` has been only given access to `bob`.

Another case might be when `String` is at `me` i.e the local machine and the rest is same as above.

I think that should cover all the cases. Rest of the tests LGTM.

Your views @AlanAboudib ?

Reply:
Oh I actually did not see Alan's comment for the tests I just talked about.